### DISCUSSION TITLE
Spurious grid pattern when rendering boundary of protected area

### DISCUSSION THREAD

**Discussion by cyril42e** [mapsforge/mapsforge#1488] [Q&A] [answered]
Hi,

I have a strange behavior when customizing a theme in order to render the contour of protected_area / national_park / nature_reserve. In addition to the actual border, a grid pattern is also rendered inside the area. This grid pattern has the same ~2km size for zoom levels 12 to 15 (which happens to correspond and be aligned with the tiles of level 14, and the same ~25km size for zoom levels 8 to 11 (which happens to correspond and be aligned with the tiles of level 10). It seems to disappear starting from zoom level 16, and seems to increase again in size below zoom level 7. The stroke size of this grid pattern does not match exactly the one used to render the contour, and varies depending on the zoom level. I have the same behavior with Mobac 2.3.3 and OruxMaps 10.6.3.

For instance with this simple theme file:
```
<?xml version="1.0" encoding="UTF-8"?>

<rendertheme xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" map-background="#F8F8F8"
    map-background-outside="#EEEEEE" version="5" xmlns="http://mapsforge.org/renderTheme"
    xsi:schemaLocation="http://mapsforge.org/renderTheme https://raw.githubusercontent.com/mapsforge/mapsforge/master/resources/renderTheme.xsd">
		
	<rule e="way" k="boundary" v="protected_area">
		<rule e="any" k="protect_class" v="pr_4">
			<line stroke="#B000ff3c" stroke-width="2.0" stroke-linecap="butt" />
		</rule>
	</rule>

</rendertheme>
```

and the [.map of the Pyrénées on OpenAndroMaps](https://ftp.gwdg.de/pub/misc/openstreetmap/openandromaps/mapsV5/europe/Pyrenees.zip), I obtain the following view of the [Réserve domaniale du Mont Valier](https://www.openstreetmap.org/#map=10/42.7888/1.1446) with Mobac at zoom levels 12 and 13:
![zoom-level-12](https://github.com/mapsforge/mapsforge/assets/7440361/f0c37887-3ecb-43a9-88ac-726c5c634bc8)
![zoom-level-13](https://github.com/mapsforge/mapsforge/assets/7440361/1a9f71e9-fc8b-46d0-b0e3-93d385e3424b)

Displaying the zoom grid shows that it corresponds to the tiles of level 14:
![zoom-level-12_grid14](https://github.com/mapsforge/mapsforge/assets/7440361/7f087ea5-6507-4e70-9bf7-a96c4e9e5fc0)

I am wondering whether there is an issue with how I wrote my theme, or there is a bug in MapsForge ? (or maybe a bug in the generation of OpenAndroMaps .map file ?)

Thanks for you insight!

**Comment by devemux86** [comment, marked as answer]
Better make the tests with software that uses the Mapsforge map engine, like Mapsforge samples or [Cruiser](https://github.com/devemux86/cruiser).
(there is desktop version)

Some of the apps may use their own rendering engine with bitmap tiles, so we cannot know how they work.

About your question, please see the relevant section in the map writer documentation:
https://github.com/mapsforge/mapsforge/blob/master/docs/Getting-Started-Map-Writer.md#known-pitfalls

You can either change the map creation or not use stroke in polygons or try an open line.

**Reply by cyril42e** [reply]
Thanks for the tips.

Do you mean that some apps don't use Mapsforge at all (but I came here because both Mobac and OruxMaps do use Mapsforge to render vector maps), or that some apps perform additional steps after Mapsforge rendering ? 

Anyway for the record I get exactly the same behavior with Cruiser:
![cruiser-level-12](https://github.com/mapsforge/mapsforge/assets/7440361/b2d38a13-1f90-4c92-a790-e4bc79d12cf2)

About the known pitfall you mention, if I try to understand correctly the different solutions you are suggesting:

1. "change the map creation": I guess that means that I should report to OpenAndroMaps that they should fix the tag-mapping that they are using?
2. "not use stroke in polygons": do you mean to simply stop trying to draw the outline of the natural parks, or doing it in another way that I don't know ?
3. "try an open line": is it possible to convert the polygon into a polyline for the rendering, without changing the map ? (other than setting force-polygon-line="true" in tag-mapping.xml, which is back to the first solution)

**Reply by devemux86** [reply]
Some apps use Mapsforge maps with their own map engine, and not Mapsforge map engine.

Quick solution: don't draw the outline with stroke in those polygons, just use the fill parameter.

**Reply by cyril42e** [reply]
> Some apps use Mapsforge maps with their own map engine, and not Mapsforge map engine.

Ok, thanks, no I came here because both Mobac and Oruxmaps do use it to render the vector maps.

> Quick solution: don't draw the outline with stroke in those polygons, just use the fill parameter.

Sure, but that's specifically what I wanted, I think it's the best way to render natural parks, that's what openstreetmap.org theme and french IGN maps do for instance. So I'm left with option 1, talking with OpenAndroMaps guys, if I don't want to regenerate the .map files myself!

Thanks for your inputs.

**Reply by cyril42e** [reply]
Also forgot to highlight that the issue is present at zoom levels from 8 to 15, not only 12-13 as mentioned in the known pitfalls, even if my illustrations happened to be at these levels.

However with semi-transparent fill, I do have the problem only at levels 12-13 as mentioned in the known pitfalls (so the fill parameter is not really a solution either).
